### PR TITLE
bsunada:Run2-hcx32 Bug fix for HcalTopology

### PR DESCRIPTION
**Pull request by bsunanda**
The bug from detId to hasedId had a bug for HB which is corrected. Also the script for Calowriter needs an additional cff.

### CODE REVIEW

**Comment by lgray** [comment]
@bsunanda Can you please fill in the pull request description with something a little more detailed than the title? Thanks.

**Comment by lgray** [comment]
@cmsbuild please test

File: CondTools/Geometry/test/calogeometrywriter.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 process = cms.Process("CaloGeometryWriter")
 process.load('CondCore.DBCommon.CondDBCommon_cfi')
-process.load('Configuration.StandardSequences.GeometryExtended_cff')
+process.load('Configuration.Geometry.GeometryExtended_cff')
+process.load('Configuration.Geometry.GeometryReco_cff')
```
Comment (ianna) [inline CondTools/Geometry/test/calogeometrywriter.py]: @bsunanda - please, remove this line. All relevant producers are include through the next one with appropriate parameters. Besides, if this file is not modified, the PR will be merged faster.

Another question: does it mean Hcal geometry records in DB are wrong and need to be recreated? Thanks

**Comment by lgray** [comment]
@cmsbuild please test

File: CondTools/Geometry/test/calogeometrywriter.py (hunk 1 above)
Comment (bsunanda) [inline CondTools/Geometry/test/calogeometrywriter.py]: Taken your advice. But StandardSequences.GeometryExtended_cff does not have initiator for recnumbering record which is needed and when I ran without Geometry.GeometryReco_cff it did not work. Hcal geometry records for run 1 and run 2 are fine - the error was only for SLHC version which was in the 62X version and was never tested.
________________________________

**Comment by ianna** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**